api/lib/domain/usecases/update-publication-session.js

```diff
@@ -0,0 +1,18 @@
+module.exports = async function updatePublicationSession({
+  sessionId,
+  toPublish,
+  certificationCourseRepository,
+}) {
+  const certificationCoursesToPublish = await certificationCourseRepository.findIdsBySessionId(sessionId);
+
+  if (certificationCoursesToPublish.length > 0) {
+    await certificationCoursesToPublish.forEach(async (certifId) => {
+      await certificationCourseRepository.update({
+        id: certifId,
+        isPublished: toPublish
+      });
```

Comment:
J'ai l'impression qu'on attend la publication d'une certif après l'autre. On pourrait tout lancer et attendre l'ensemble, ou bien envoyer par paquet

Reply:
La méthode [forEach](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/forEach) retourne `undefined` donc les fonctions asynchrone sont de fait appelé en parallèle mais la manière dont le code est écrit laisse sous entendre qu'on attendant à chaque itération.

Si on veut garder ce comportement parallèle alors on peut enlever les mot-clés `async/await` aux lignes 9 et 10. Si au contraire on veut les exécuter une à une alors il faut passer sur une implémentation à base de boucle `for`. Si on veut les attendre toutes d'un coup on peut utiliser `map` avec un `Promise.all` sur son résultat.

Quelle est l'intention ici ?